src/main/java/com/aws/greengrass/util/platforms/unix/LinuxPlatform.java

```diff
@@ -0,0 +1,143 @@
+        try {
+            if (resourceLimit.containsKey(MEMORY_KEY)) {
+                long memoryLimitInKB = (long) resourceLimit.get(MEMORY_KEY);
+
+                // TODO: add input validation
+                String memoryLimit = Long.toString(memoryLimitInKB * 1024);
+                Files.write(LinuxCgroup.Memory.getSubsystemComponentPath(componentName).resolve(CGROUP_MEMORY_LIMITS),
+                        memoryLimit.getBytes());
+            }
+
+            if (resourceLimit.containsKey(CPU_KEY)) {
+                double cpu = (double) resourceLimit.get(CPU_KEY);
+
+                byte[] content = Files.readAllBytes(
+                        LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cpu.cfs_period_us"));
+                int cpuPeriodUs = Integer.parseInt(new String(content));
+
+                int cpuQuotaUs = (int) (cpuPeriodUs * cpu);
+                String cpuQuotaUsStr = Integer.toString(cpuQuotaUs);
+
+                Files.write(LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cpu.cfs_quota_us"),
+                        cpuQuotaUsStr.getBytes());
+            }
+
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName)
+                    .log("Failed to apply resource limits");
+        }
+    }
+
+    public void addProcessToResourceManager(String componentName, Process process) {
+        if (process == null) {
+            logger.atError().log("The process doesn't exist");
+            return;
+        }
+        int pid = PidUtil.getPid(process);
+        try {
+            Files.write(LinuxCgroup.Memory.getSubsystemComponentPath(componentName).resolve("cgroup.procs"),
+                    Integer.toString(pid).getBytes());
+            Files.write(LinuxCgroup.CPU.getSubsystemComponentPath(componentName).resolve("cgroup.procs"), Integer
+                    .toString(pid).getBytes());
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName).kv("processId", pid)
+                    .log("Failed to add pid to the cgroup");
+        }
+    }
+
+    private void runCmdWithRoot(String command) throws IOException, InterruptedException {
+        String[] cmd = getUserDecorator().withUser(getPrivilegedUser()).decorate(command);
+
+        logger.atDebug().log("run command with root: {}", command);
+        Process proc = Runtime.getRuntime().exec(cmd);
+        proc.waitFor();
+        if (proc.exitValue() != 0) {
+            logger.atWarn().kv(STDOUT, inputStreamToString(proc.getInputStream()))
+                    .kv(STDERR, inputStreamToString(proc.getErrorStream()))
+                    .log("Fail to run command with root: {}", command);
+        }
+    }
+
+    private Set<String> getMountedPaths() throws IOException {
+        Set<String> mountedPaths = new HashSet<>();
+
+        Path procMountsPath = Paths.get("/proc/self/mounts");
```

Comment:
how does this work? are `self/mounts` different from what I would get if I ran `mount` to list all the mounts on the system?